src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryLoggerExtensions.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Antiforgery.Internal
+{
+    internal static class DefaultAntiforgeryLoggerExtensions
+    {
+        private static readonly Action<ILogger, string, Exception> _validationFailed;
+        private static readonly Action<ILogger, Exception> _validating;
+        private static readonly Action<ILogger, string, Exception> _missingCookieToken;
+        private static readonly Action<ILogger, string, Exception> _missingFormToken;
+        private static readonly Action<ILogger, string, Exception> _missingHeaderToken;
+        private static readonly Action<ILogger, string, string, Exception> _missingRequestToken;
+
+        static DefaultAntiforgeryLoggerExtensions()
+        {
+            _validationFailed = LoggerMessage.Define<string>(
+                LogLevel.Warning,
+                1,
+                "Antiforgery validation failed with this message '{Message}'");
+            _validating = LoggerMessage.Define(
+                LogLevel.Information,
+                2,
+                "Antiforgery is attempting to validate a request.");
+            _missingCookieToken = LoggerMessage.Define<string>(
+                LogLevel.Warning,
```

Comment:
So we should use a different logging level for the same event in `IsRequestValidAsync()`? If yes, sounds good to me.